lib/context.js

```diff
@@ -109,6 +109,9 @@ const proto = module.exports = {
     // to node-style callbacks.
     if (null == err) return;
 
+    // ENOENT support
+    if ('ENOENT' == err.code) err = createError(404);
```

Comment:
why move this to front? Also we should not change the original `err` here

Reply:
An ENOENT error is NOT a instance of Error, so it will be considered a non-error.

![image](https://user-images.githubusercontent.com/17722900/45461253-91f90a80-b734-11e8-819f-fc9458a76115.png)

And server will reply a 500 instead of 404.

![image](https://user-images.githubusercontent.com/17722900/45461390-590d6580-b735-11e8-8562-c6d2e4cc38ff.png)